src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -356,6 +386,17 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary=False):
                                 "archivestatus": archStatus,
                                 "fwjr": jsonFWJR,
                                 "type": "fwjr"}
+                if countDocs >= self.fwjrdatabase.getQueueSize():
+                    try:
+                        self.fwjrdatabase.commit(callback=discardConflictingDocument)
+                    except CouchRequestTooLargeError as exc:
+                        msg = "Failed to commit bulk of framework job report to CouchDB."
+                        msg += f" Details: {str(exc)}"
+                        logging.warning(msg)
+                        shrinkLargeFJR(self.fwjrdatabase, 8e6)
+                        # now all the documents should fit in
```

Comment:
you must add second check here to confirm that all docs are below CouchDB max doc limit, otherwise you cannot guarantee that after shrinkage document will still fit.

Reply:
I don't understand this comment. `shrinkLargeFJR` function will actually parse all the documents in the database queue and shrink all those that are actually larger than the limit. I don't see why a second check would be required here. Please clarify

Reply:
see my comment above in this function which requires second check of doc size after `doc['fwjr'] = dict()` assignment.